src/include/daos_srv/daos_server.h

```diff
@@ -175,6 +240,8 @@ struct dss_module_info {
 	/* the cart context id */
 	int			dmi_ctx_id;
 	d_list_t		dmi_dtx_batched_list;
+	/* the profile information */
+	struct srv_profile	*dmi_sp;
```

Comment:
Suggestion: Don't rely on server TLS code, just directly use pthread_set/getspecific(), so that it can be used for profiling standalone VOS and client stack.

Reply:
hmm, this might work. I will check. thanks.